lib/paper_trail/record_trail.rb

```diff
@@ -150,9 +150,13 @@ def source_version
     # This is an "update" event. That is, we record the same data we would in
     # the case of a normal AR `update`.
     def save_with_version(in_after_callback: false, **options)
-      ::PaperTrail.request(enabled: false) do
-        @record.save(**options)
-      end
+      save_result =
+        ::PaperTrail.request(enabled: false) do
+          @record.save(**options)
+        end
+
+      return save_result unless enabled?
```

Comment:
I don't think `enabled?` is an expensive method, but it will be called by `record_update`, so it might be wasteful to call it a second time here. What do you think of the following?

```ruby
def save_with_version(in_after_callback: false, **options)
  save_result = ::PaperTrail.request(enabled: false) do
    @record.save(**options)
  end
  record_update(force: true, in_after_callback: in_after_callback, is_touch: false)
  save_result
end
```

Actually, this raises some other questions. 

- Why are we calling `record_update` if the `save` failed? That seems wrong.
- Perhaps `save_with_version` should only return true if both `save` and `record_update` succeed.

Reply:
These probably aren't questions you _wanted_ me to ask :), but if we're going to make a breaking change, I think we should address these questions holistically.

Reply:
No worries about asking any questions 😅 

I actually asked those questions myself. My goal was to minimize the change that I'm doing to the API while achieving the goal I wanted. The problem I see with your proposed solution is that it will always return a boolean even if version updates are enabled (currently it would return the created version object).

A previous version that I had in mind was like this:

```ruby
def save_with_version(in_after_callback: false, **options)
  save_result = ::PaperTrail.request(enabled: false) do
    @record.save(**options)
  end
  save_result && record_update(force: true, in_after_callback: in_after_callback, is_touch: false)
end
```

I did not use this variant because it changes the behavior if saving failed. However I can totally see this being better (as you mentioned as well). I'm just not sure if other people might depend on this. What do you think? I'm also not sure what should be returned if version creation failed. I think it would return `nil` currently but I did not check that yet.

Reply:
Yeah, it seems wrong to save a `Version` record when `@record.save` failed. That seems like a bug to me. Do you mind fixing it, please, while we're in here? I know it's a bit out of your original scope.

> The problem I see with your proposed solution is that it will always return a boolean ..

I'd think I'd like it to always return a boolean, because that's how Rails' `save` works. You're right that we should treat this as a breaking change. We can increment the major version number (Not in this PR please, I'll increment it at release-time)

WDYT?

Reply:
Hey @jaredbeck! Is there anything I can do to move this forward? Would love to get consent on the change so that we can get this merged as soon as possible.